src/main/scala/com/microsoft/hyperspace/index/IndexLogEntry.scala

```diff
@@ -326,6 +326,19 @@ object LogicalPlanFingerprint {
   case class Properties(signatures: Seq[Signature])
 }
 
+/**
+ * Updated source data since lst time derived dataset was updated. Note that index data does not
+ * cover appendedFiles and deleted Files.
+ *
+ * @param appendedFiles Appended files.
+ * @param deletedFiles Deleted files.
+ * @param latestSignatures Latest signatures including appendedFiles and deletedFiles.
+ */
+case class Update(
+  latestSignatures: Seq[Signature],
```

Comment:
I thought about this field, and can we make this as `fingerprint: LogicalPlanFingerprint`? This goes against my previous comment about making this `Update` generic. Let me explain.

Whenever you see "kind" and "properties", that structure is supposed to be a "plug-in" point. For example, the "Source" has "SparkPlan", but it can be replaced with different sources "SqlDW", etc. It's easier to see if you look at the JSON representation. We never got a chance to work on the parser to support these scenarios and resorted to the `case class` to make reading/writing JSON easy for the time being.

Back to `Update`, I think the spec owner can decide what to do with the fingerprint for `Update`. In our case, it will be a `LogicalPlanFingerprint`. This also aligns with `Content` having a `fingerprint`. Does this make sense? We can update the comment as follows:
```scala
/**
 * Captures any HDFS updates.
 *
 * @note 'fingerprint' shouldn't be tied to [[LogicalPlanFingerprint]], but it's a free-form
 *       class, meaning it has "kind" and "properties" so that different classes can be plugged in.
 *       Thus, 'fingerprint' is still a generic field in terms of the metadata log specification.
 *
 * @param fingerprint Fingerprint of the update.
 * @param appendedFiles Appended files.
 * @param deletedFiles Deleted files.
 */
case class Update(
  fingerprint: LogicalPlanFingerprint,
  appendedFiles: Option[Content] = None,
  deletedFiles: Option[Content] = None)
```

Reply:
That makes sense. Thanks for the suggestion :)